### PR TITLE
add ring signature guide

### PR DESCRIPTION
**Pull request by tinnus-napbus**
ring signature voting app lightning guide

this has been restructured as discussed with @jalehman so the first part just speedruns creating the desk and adding the different files, and then after that there's additional commentary on how it works. It's maybe too light on explanation & also it's maybe not ideal to have such code explanations separate to the actual code. I didn't think I should repost the code below though, so idk.

I have added cards to the front-page and guides page for it. It needs a new icon though, currently it's the same as the chat guide. @gordoncc can you create a voting-related icon for this?

Also this rebuilds the package-lock file so the new collapsible & copyable codeblocks will work

### CODE REVIEW

File: content/guides/quickstart/voting-guide.md [hunk 1]
```diff
@@ -0,0 +1,1230 @@
...
+can be verified to have come from a group member and duplicate votes can be
+detected, but it cannot be determined who voted for what. The finished app will
+look like this:
+
+![tally screenshot](https://media.urbit.org/guides/quickstart/voting-app-guide/tally-screenshot.png)
+
+The front-end of the app will be written in
+[Sail](/reference/glossary/sailudon), Urbit's XML language built into the Hoon
+compiler. Using Sail means we don't need to create a separate React front-end,
+and can instead serve pages directly from our back-end.
```
Comment (sigilante) [inline content/guides/quickstart/voting-guide.md:20]: “This works well for static pages but a full JS-enabled front-end would be preferred for a dynamic page.”

File: content/guides/quickstart/voting-guide.md [hunk 2]
```diff
@@ -0,0 +1,1230 @@
...
+
+```bash
+./urbit -F zod
+```
+
+It'll take a couple of minutes to boot up, and then it'll take us to the Dojo.
+
+### Dependencies
+
+Once in the Dojo, let's mount a couple of desks so their files can be accessed
```
Comment (sigilante) [inline content/guides/quickstart/voting-guide.md:74]: (as indicated by the prompt `dojo>`)

File: content/guides/quickstart/voting-guide.md [hunk 3]
```diff
@@ -0,0 +1,1230 @@
...
+from the host OS. We can do this with the `|mount` command:
+
+```
+|mount %base
+|mount %garden
+|mount %landscape
+```
+
+With those mounted, switch back to a normal shell. We'll create a folder to
```
Comment (sigilante) [inline content/guides/quickstart/voting-guide.md:83]: (in another terminal window)

File: content/guides/quickstart/voting-guide.md [hunk 4]
```diff
@@ -0,0 +1,1230 @@
...
+develop our app in, and then we'll copy a few files across that our app will
+depend on:
+
+```bash
```
Comment (sigilante) [inline content/guides/quickstart/voting-guide.md:87]: Didn't Matilde say we can annotate to make these copyable now?

File: content/guides/quickstart/voting-guide.md [hunk 5]
```diff
@@ -0,0 +1,1230 @@
...
+  |=  =ship
+  ?|  =(our.bol ship)
+      ?=(?(%czar %king %duke) (clan:title ship))
+  ==
+--
+```
+
+### Marks
+
+Marks are Urbit's version of filetypes/MIME types. We need to define a mark for
```
Comment (sigilante) [inline content/guides/quickstart/voting-guide.md:590]: “but they include transformation rules as well”

File: content/guides/quickstart/voting-guide.md [hunk 6]
```diff
@@ -0,0 +1,1230 @@
...
+  --
+++  grad  %noun
+--
+```
+
+### Front-end
+
+We could have put the front-end code directly in our Gall agent, but it tends to
+be quite large so it's convenient to have it in a separate file and just import
+it.
```
Comment (sigilante) [inline content/guides/quickstart/voting-guide.md:637]: “Most of this file consists of Sail code, which is the internal HTML representation, similar to other server-side renderings like Clojure's Hiccup.”

File: content/guides/quickstart/voting-guide.md [hunk 7]
```diff
@@ -0,0 +1,1230 @@
...
+  license+'MIT'
+  base+'tally'
+  site+/tally
+==
+```
+
+### Put it together
+
+Our app is now complete, so let's try it out. In the Dojo of our fake ~zod,
+we'll create a new desk by forking from an existing one:
```
Comment (sigilante) [inline content/guides/quickstart/voting-guide.md:1023]: “(filesystem branch)”

File: content/guides/quickstart/voting-guide.md [hunk 8]
```diff
@@ -0,0 +1,1230 @@
...
+a real ship. To do that, we can just repeat the steps in this section with our
+actual ship. On the live network, we can also publish the app so others can
+install it from us. To do so, just run the following command:
+
+```
+:treaty|publish %tally
+```
+
+Now our friends will be able to install it with `|install <our ship> %tally`
+or by searching for `<our ship>` on their ship's homescreen.
```
Comment (sigilante) [inline content/guides/quickstart/voting-guide.md:1065]: “Ships are intended to run continuously, but fake ships for development are frequently stopped between sessions.  Use `|exit` to shut down your fake ship.”

**Review by sigilante — APPROVED** [review]
Modulo these suggestions, looks fantastic.

**Review by jalehman — APPROVED** [review]
Looks great! Just a couple of small copy fixes that I noticed.

**Pull request merged**